k4FWCore/include/k4FWCore/Transformer.h

```diff
@@ -126,6 +125,17 @@ namespace k4FWCore {
         }
       }
 
+      auto inputLocations(int i) const {
+        if (i >= sizeof...(In)) {
+          throw GaudiException("Called inputLocations with an index out of range", "Consumer", StatusCode::FAILURE);
+        }
+        return m_inputLocations[i] | std::views::transform([](const DataObjID& id) -> const auto& { return id.key(); });
+      }
```

Comment:
The check could happen at compile time since the number of input/output arguments is known. I'm not sure if there is a point in allowing a runtime argument to this. If there is then probably then should be a function to get the number of inputs/outputs (Gaudi functional algorithms have `InputLocationSize` to get `sizeof...(In)`)
```suggestion
      template<size_t i>
      auto inputLocations() const {
        static_assert(i < sizeof...(In), "Index must be smaller than the number of inputs of the algorithm");
        return m_inputLocations[i] | std::views::transform([](const DataObjID& id) -> const auto& { return id.key(); });
      }
    };
```

Reply:
But the index is not known at compile time, for example, when running inside a loop or any other logic that someone may use for indexing so it's not possible to do a `static_assert`. For the other comment I don't think we need a function to call `sizeof...` which is short enough and there is not a lot of use case for that from an algorithm given that it is known, no?

Reply:
We could demand the index at the compile time (template argument) if we don't want people to write loops over arguments. If someone wrote
```cpp
vector<MCParticleCollection> operator()(const vector<const MCParticleCollection*>&,  const vector<const SimTrackerHitCollection*>&)
```
 then I'd expect they should be able to understand "at compile time" that `vector<MCParticleCollection*>` is the first argument  and `vector<SimTrackerHitCollection*>` is the second argument. Using an index is just a convention

If writing loops over arguments is allowed then I think we should provide some way to get the bound. `sizeof...(In)` looks to me a bit too much of an implementation detail, since the algorithm developers probably don't have to use `In` directly at all. 
```cpp
for(size_t i = 0; i != sizeof...(In); ++i){
  auto locations = inputLocations(i);
}
```
looks a bit more daunting than
```cpp
for(size_t i = 0; i != inputLocationsSize(); ++i){
  auto locations = inputLocations(i);
}
```

Reply:
Ok, I have included an `inputLocationsSize` and `outputLocationsSize` methods.